src/rope.rs

```diff
@@ -2052,9 +2052,41 @@ impl std::cmp::PartialOrd<Rope> for Rope {
 
 impl std::hash::Hash for Rope {
     fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        // we can not simply call state.write(chunk) for each chunk
+        // because Hasher.write only produces the same input if the
+        // calls match exactly. See documentation of hash and hast_02 test below
+        // so we have to perform some tricks here to ensure that the same
+        // calls to write are produce no matter the chunk layout
+        let mut accumulator = [0u8; MIN_BYTES];
```

Comment:
I think it would make more sense for the accumulator buffer to just have its own fixed size, rather than being based on `MIN_BYTES`.  I'm not sure what size would be best, but we can play around with it to see what performs best.

Reply:
I will setup some benchmarks. The problem is that the performance highly depends on the exact hasher implementation. I think we generally want a a multiple of `16` here (which MIN_BYTES is) that is smaller then most chunks so that the `chunk.len() < MIN_BYTES - accumulator_pos` branch can get good branch prediction. So `MIN_BYTES` might just be a sweetspot but I guess I will look into benchmarking. Another reason `MIN_BYTES` is attractive here is that it is fixed to the chunksize so if that ever changes we won't have to redetermine the ideal buffer size

Reply:
I have added a seperate constant for `BUFFER_SIZE` from your comment below it sounds like you will change the implementation anyway so I didn't invest the time to benchmark it and just set it to `BYTES_MIN`